src/roiextractors/extractors/suite2p/suite2psegmentationextractor.py

```diff
@@ -56,12 +68,27 @@ def __init__(
         self.plane_no = plane_no
         self.folder_path = Path(folder_path)
 
-        self.stat = self._load_npy("stat.npy")
-        self._roi_response_raw = self._load_npy("F.npy", mmap_mode="r").T
-        self._roi_response_neuropil = self._load_npy("Fneu.npy", mmap_mode="r").T
-        self._roi_response_deconvolved = self._load_npy("spks.npy", mmap_mode="r").T
-        self.iscell = self._load_npy("iscell.npy", mmap_mode="r")
-        self.ops = self._load_npy("ops.npy").item()
+        self._search_plane_subdirectory = search_plane_subdirectory
+
+        def try_load_npy(filename, mmap_mode=None, fn_transform=lambda x: x):
+            """
+            This function allows for incomplete import of files.
+            """
+            try:
+                return fn_transform(self._load_npy(filename, mmap_mode=mmap_mode))
+            except FileNotFoundError:
+                if allow_incomplete_import:
+                    warnings.warn(f"File {filename} not found.") if warn_missing_files else None
+                    return None
+                else:
+                    raise FileNotFoundError(f"File {filename} not found.")
```

Comment:
So, if I understand your issue correctly, the problem that you encounter trying to use our reader is that your example of Suite2P output does not have all of these files present in each subfolder (by plane)?

If so, then I think it's fine to refactor this to only load the subset of available traces, and no need to error, warn, or have it be flag controllable (and just always load everything it can)

This would simplify the code here a lot; I'll suggest a simple ternary structure below for this

Reply:
> your example of Suite2P output does not have all of these files present in each subfolder (by plane)?
Correct. I and others often work with only stat.npy and/or ops.npy files for doing ROI tracking across sessions.

Reply:
they are also generally not in strict directory patterns like 'plane0' etc.

Reply:
> they are also generally not in strict directory patterns like 'plane0' etc.


Well, this is one part that is fairly important since we interpret all of this as being enforceable as a part of Suite2P format - are you saying that as you run Suite2P, it only saves the individual files and does not actually save a sessions worth of data using the `plane{integer}` subfolder format?

Reply:
> are you saying that as you run Suite2P, it only saves the individual files and does not actually save a sessions worth of data using the plane{integer} subfolder format?

No. However, it is not uncommon for these directories to have their names changed or to copy only data that are necessary for subsequent steps. Given that stat.npy and ops.npy are very small compared to F.npy and other files, they are often moved without the others.

I'd say that this is not a huge deal and I think having strong opinions on directory structures is probably wise. Though, allowing incomplete file ingestion is pretty critical still.

Reply:
> I'd say that this is not a huge deal and I think having strong opinions on directory structures is probably wise. Though, allowing incomplete file ingestion is pretty critical still.

👍  OK sounds good, thanks for clarifying!